shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1696,116 +1698,117 @@ - (void)startKeyBoardAnimation:(NSTimeInterval)duration {
+        strongSelf.keyboardAnimationView.frame = CGRectMake(0, self.targetViewInsetBottom, 0, 0);
 
         // Setup keyboard animation interpolation.
         CAAnimation* keyboardAnimation =
-            [[self keyboardAnimationView].layer animationForKey:@"position"];
-        [self setUpKeyboardSpringAnimationIfNeeded:keyboardAnimation];
+            [strongSelf.keyboardAnimationView.layer animationForKey:@"position"];
+        [strongSelf setUpKeyboardSpringAnimationIfNeeded:keyboardAnimation];
       }
       completion:^(BOOL finished) {
         if (_keyboardAnimationVSyncClient == currentVsyncClient) {
+          FlutterViewController* strongSelf = weakSelf;
+          if (!strongSelf) {
+            return;
+          }
+
           // Indicates the vsync client captured by this block is the original one, which also
           // indicates the animation has not been interrupted from its beginning. Moreover,
           // indicates the animation is over and there is no more to execute.
-          [self invalidateKeyboardAnimationVSyncClient];
-          [self removeKeyboardAnimationView];
-          [self ensureViewportMetricsIsCorrect];
+          [strongSelf invalidateKeyboardAnimationVSyncClient];
+          [strongSelf removeKeyboardAnimationView];
+          [strongSelf ensureViewportMetricsIsCorrect];
         }
       }];
 }
 
 - (void)setUpKeyboardSpringAnimationIfNeeded:(CAAnimation*)keyboardAnimation {
   // If keyboard animation is null or not a spring animation, fallback to DisplayLink tracking.
   if (keyboardAnimation == nil || ![keyboardAnimation isKindOfClass:[CASpringAnimation class]]) {
-    _keyboardSpringAnimation.reset();
+    _keyboardSpringAnimation = nil;
     return;
   }
 
   // Setup keyboard spring animation details for spring curve animation calculation.
   CASpringAnimation* keyboardCASpringAnimation = (CASpringAnimation*)keyboardAnimation;
-  _keyboardSpringAnimation.reset([[SpringAnimation alloc]
-      initWithStiffness:keyboardCASpringAnimation.stiffness
-                damping:keyboardCASpringAnimation.damping
-                   mass:keyboardCASpringAnimation.mass
-        initialVelocity:keyboardCASpringAnimation.initialVelocity
-              fromValue:self.originalViewInsetBottom
-                toValue:self.targetViewInsetBottom]);
+  _keyboardSpringAnimation =
+      [[SpringAnimation alloc] initWithStiffness:keyboardCASpringAnimation.stiffness
+                                         damping:keyboardCASpringAnimation.damping
+                                            mass:keyboardCASpringAnimation.mass
+                                 initialVelocity:keyboardCASpringAnimation.initialVelocity
+                                       fromValue:self.originalViewInsetBottom
+                                         toValue:self.targetViewInsetBottom];
+}
+
+- (void)handleKeyboardAnimationCallback:(fml::TimePoint)targetTime {
```

Comment:
`handleKeyboardAnimationCallbackWithTargetTime:` (since the param is a target time, not a callback, and the signature should name/describe the param).

Reply:
Done.